<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Dimension reduction</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/anchor-sections-1.0/anchor-sections.css" rel="stylesheet" />
<script src="site_libs/anchor-sections-1.0/anchor-sections.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Miller lab data</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Version control
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="version_control.html">Getting started</a>
    </li>
    <li>
      <a href="git_workflow.html">Git workflow</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Using R
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="pre_reqs.html">But first</a>
    </li>
    <li>
      <a href="lets_r.html">Let's R 1</a>
    </li>
    <li>
      <a href="lets_r_2.html">Let's R 2</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Metabolomics
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="worm_metabolomics_1.html">Worm metabolomics 1</a>
    </li>
    <li>
      <a href="worm_metabolomics_2.html">Worm metabolomics 2</a>
    </li>
    <li>
      <a href="worm_metabolomics_3.html">Worm metabolomics 3</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Other resources
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="project_organization.html">Project organization</a>
    </li>
    <li>
      <a href="feature_extraction.html">Metabolomics data extraction</a>
    </li>
    <li>
      <a href="ggbiplot.html">ggbiplot</a>
    </li>
    <li>
      <a href="dim_red.html">Dimension reduction</a>
    </li>
  </ul>
</li>
<li>
  <a href="files.html">Files of interest</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">Dimension reduction</h1>

</div>


<p>Dimensionn reduction is often used in omics data analysis. While the excerpt below relates to dimension reduction in the context of high-resolution mass spec data, here are other resources you can use for understanding dimension reduction:</p>
<ul>
<li><a href="https://www.mailman.columbia.edu/research/population-health-methods/principal-components-analysis">Mailman: PCA</a><br />
</li>
<li><a href="http://web.stanford.edu/class/bios221/PCA_Slides.html">Stanford: multivariate analysis</a></li>
</ul>
<p><em>Taken from Kalia et al, Unsupervised dimensionality reduction for exposome research, under review, 2020</em></p>
<p>&quot;Exposomic data from HRMS are high-dimensional and complex. When untargeted methods are used to characterize the exposome, the number of chemical signals can be greater than 100,000 <span class="citation">(Vermeulen et al. 2020)</span>. These data tables represent the primary input for bioinformatic and biostatistical analysis to evaluate biological meaning and to determine exogenously derived chemicals that may be of relevance to an outcome of interest. Since detected analytes are not annotated a priori, the results from HRMS contain information on each detected ion (a feature). Each feature is characterized by its mass-to-charge ratio, the retention time at which the compound eluted from the chromatographic column, and its abundance in the sample. Some features may arise from the same parent compound but during ionization may lead to formation of multiple ions with different masses with the same retention times. Therefore, the feature table may contain degenerate features which will be highly correlated since they arise from the same parent compound. Thus, correlation is present not only due to exposure sources and biological processes, but also arises from the analysis itself <span class="citation">(Alonso, Marsal, and Julià 2015)</span>. This forces researchers to deal with the “curse of dimensionality” and to capture the essence of the data generated <span class="citation">(Bellman 1966)</span>. Given the high correlations across chemical exposures, single chemical association studies are not appropriate, with implications for health that are missed using a “candidate” approach. Furthermore, high correlation compounds correction for multiple testing, decreasing study power and inflating the potential for false negatives.</p>
<p>Data dimensionality reduction approaches produce a manageable number of variables, allow for better visualization, remove redundant and uninformative variables, and reduce computational burden <span class="citation">(Xue, Stahura, and Bajorath 2004)</span>. Several techniques have been described that reduce the size of the data table while minimizing loss of information, describing the essence of the data generated. These include unsupervised and supervised methods. In supervised methods, the outcome of interest informs the dimensionality reduction solution. These methods are used for feature selection and include—but are not limited to—different forms of penalized regression: least absolute shrinkage and selection operator (LASSO) <span class="citation">(Tibshirani 1996)</span>, ridge regression <span class="citation">(Hoerl and Kennard 1970)</span>, and elastic net regression <span class="citation">(Zou and Hastie 2005)</span>, and different modifications of partial least squares regression <span class="citation">(Lê Cao, Boitard, and Besse 2011, <span class="citation">Nguyen and Rocke (2002)</span>)</span>. Unsupervised methods do not take the outcome of interest under consideration during feature extraction. The goal in unsupervised dimensionality reduction is to discover the underlying structure in the data. These methods are tuned for pattern recognition that can aid in data visualization, data exploration, and uncovering latent variables. A popular method for unsupervised linear transformation of data is principal components analysis. Different unsupervised methods also exist for non-linear data transformations <span class="citation">(Bartenhagen et al. 2010)</span> like the kernel PCA <span class="citation">(Smola 1999)</span>, isomap <span class="citation">(Tenenbaum, De Silva, and Langford 2000)</span> and autoencoders <span class="citation">(Kingma and Welling 2014)</span>.</p>
<div id="principal-components-analysis-pca" class="section level4">
<h4>Principal components analysis (PCA)</h4>
<p>Principal components analysis is one of the most commonly used approaches for dimensionality reduction. The method uses an orthogonal transformation to convert a set of observations of possibly correlated variables into a set of linearly uncorrelated variables called principal components. The first component explains the most variance in the data and each succeeding component has the highest variance possible under the constraint that it is orthogonal, i.e., independent, to the preceding components <span class="citation">(Jolliffe 2002)</span>. The method does not reduce the number of variables, m variables produce m components. The analyst chooses the number of components to include in analyses based on some a priori defined criterion/a, e.g., looking at the scree plot, selecting components with eigenvalues above one, or selecting the number of components that explain a pre-specified proportion of the variance in the data, e.g., at least 75%. Since PCA forces orthogonality between components, it imposes a rigid structure <span class="citation">(Liland 2011)</span>. The alternating least squares variant of PCA, independent component analysis, is more successful in dealing with this rigidity but has less compression in the first components <span class="citation">(Comon 1994)</span>.&quot;</p>
<p><img src = "ppt_images/dim_red_visual.png" style="width:65%; border:2px solid; margin-right: 20px"></p>
<p><em>Visual representation of dimension reduction using PCA, FA, and NMF.</em></p>
<hr />
</div>
<div id="references" class="section level3 unnumbered">
<h3>References</h3>
<div id="refs" class="references">
<div id="ref-alonso_analytical_2015">
<p>Alonso, Arnald, Sara Marsal, and Antonio Julià. 2015. “Analytical Methods in Untargeted Metabolomics: State of the Art in 2015.” <em>Frontiers in Bioengineering and Biotechnology</em> 3. doi:<a href="https://doi.org/10.3389/fbioe.2015.00023">10.3389/fbioe.2015.00023</a>.</p>
</div>
<div id="ref-bartenhagen_comparative_2010">
<p>Bartenhagen, Christoph, Hans-Ulrich Klein, Christian Ruckert, Xiaoyi Jiang, and Martin Dugas. 2010. “Comparative Study of Unsupervised Dimension Reduction Techniques for the Visualization of Microarray Gene Expression Data.” <em>BMC Bioinformatics</em> 11 (1): 567. doi:<a href="https://doi.org/10.1186/1471-2105-11-567">10.1186/1471-2105-11-567</a>.</p>
</div>
<div id="ref-bellman_dynamic_1966">
<p>Bellman, Richard. 1966. “Dynamic Programming.” <em>Science</em> 153 (3731): 34–37.</p>
</div>
<div id="ref-comon_independent_1994">
<p>Comon, Pierre. 1994. “Independent Component Analysis, A New Concept?” <em>Signal Processing</em>, Higher Order Statistics, 36 (3): 287–314. doi:<a href="https://doi.org/10.1016/0165-1684(94)90029-9">10.1016/0165-1684(94)90029-9</a>.</p>
</div>
<div id="ref-hoerl_ridge_1970">
<p>Hoerl, Arthur E., and Robert W. Kennard. 1970. “Ridge Regression: Biased Estimation for Nonorthogonal Problems.” <em>Technometrics</em> 12 (1): 55–67.</p>
</div>
<div id="ref-jolliffe_principal_2002">
<p>Jolliffe, I. T. 2002. <em>Principal Component Analysis</em>. 2nd ed. Springer Series in Statistics. New York: Springer-Verlag. doi:<a href="https://doi.org/10.1007/b98835">10.1007/b98835</a>.</p>
</div>
<div id="ref-kingma_auto-encoding_2014">
<p>Kingma, Diederik P., and Max Welling. 2014. “Auto-Encoding Variational Bayes.” <em>arXiv:1312.6114 [Cs, Stat]</em>, May. <a href="http://arxiv.org/abs/1312.6114" class="uri">http://arxiv.org/abs/1312.6114</a>.</p>
</div>
<div id="ref-le_cao_sparse_2011">
<p>Lê Cao, Kim-Anh, Simon Boitard, and Philippe Besse. 2011. “Sparse PLS Discriminant Analysis: Biologically Relevant Feature Selection and Graphical Displays for Multiclass Problems.” <em>BMC Bioinformatics</em> 12 (1): 253. doi:<a href="https://doi.org/10.1186/1471-2105-12-253">10.1186/1471-2105-12-253</a>.</p>
</div>
<div id="ref-liland_multivariate_2011">
<p>Liland, Kristian Hovde. 2011. “Multivariate Methods in Metabolomics – from Pre-Processing to Dimension Reduction and Statistical Analysis.” <em>TrAC Trends in Analytical Chemistry</em> 30 (6): 827–41. doi:<a href="https://doi.org/10.1016/j.trac.2011.02.007">10.1016/j.trac.2011.02.007</a>.</p>
</div>
<div id="ref-nguyen_tumor_2002">
<p>Nguyen, Danh V., and David M. Rocke. 2002. “Tumor Classification by Partial Least Squares Using Microarray Gene Expression Data.” <em>Bioinformatics</em> 18 (1): 39–50. doi:<a href="https://doi.org/10.1093/bioinformatics/18.1.39">10.1093/bioinformatics/18.1.39</a>.</p>
</div>
<div id="ref-smola_advances_1999">
<p>Smola, Alex. 1999. “Advances in Kernel Methods: Support Vector Learning.”</p>
</div>
<div id="ref-tenenbaum_global_2000">
<p>Tenenbaum, Joshua B., Vin De Silva, and John C. Langford. 2000. “A Global Geometric Framework for Nonlinear Dimensionality Reduction.” <em>Science</em> 290 (5500): 2319–23.</p>
</div>
<div id="ref-tibshirani_regression_1996">
<p>Tibshirani, Robert. 1996. “Regression Shrinkage and Selection via the Lasso.” <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> 58 (1): 267–88. <a href="https://www.jstor.org/stable/2346178" class="uri">https://www.jstor.org/stable/2346178</a>.</p>
</div>
<div id="ref-vermeulen_exposome_2020">
<p>Vermeulen, Roel, Emma L. Schymanski, Albert-László Barabási, and Gary W. Miller. 2020. “The Exposome and Health: Where Chemistry Meets Biology.” <em>Science</em> 367 (6476): 392–96. doi:<a href="https://doi.org/10.1126/science.aay3164">10.1126/science.aay3164</a>.</p>
</div>
<div id="ref-xue_cell-based_2004">
<p>Xue, Ling, Florence L. Stahura, and Jürgen Bajorath. 2004. “Cell-Based Partitioning.” In <em>Chemoinformatics: Concepts, Methods, and Tools for Drug Discovery</em>, edited by Jürgen Bajorath, 279–89. Methods in Molecular Biology™. Totowa, NJ: Humana Press. doi:<a href="https://doi.org/10.1385/1-59259-802-1:279">10.1385/1-59259-802-1:279</a>.</p>
</div>
<div id="ref-zou_regularization_2005">
<p>Zou, Hui, and Trevor Hastie. 2005. “Regularization and Variable Selection via the Elastic Net.” <em>Journal of the Royal Statistical Society: Series B (Statistical Methodology)</em> 67 (2): 301–20.</p>
</div>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
